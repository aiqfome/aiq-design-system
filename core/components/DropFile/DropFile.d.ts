import React from 'react';
interface TypeTranslate {
    message: {
        row1: string;
        row2: string;
    };
    errorSize: string;
    errorType: string;
}
export interface Props {
    dataMaxSize?: number;
    onChange?: (e: any) => void;
    initImage?: string;
    name?: string;
    translate?: TypeTranslate;
    width?: number | string;
    maxWidth?: number | string;
    errorMessage?: string;
    errorForm?: boolean;
}
export declare const DropFile: React.ForwardRefExoticComponent<Props & React.RefAttributes<HTMLInputElement>>;
export {};
//# sourceMappingURL=DropFile.d.ts.map