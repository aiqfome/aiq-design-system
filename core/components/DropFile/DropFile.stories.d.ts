import React from 'react';
declare const _default: {
    component: React.ForwardRefExoticComponent<import("./DropFile").Props & React.RefAttributes<HTMLInputElement>>;
    title: string;
};
export default _default;
export declare const Basic: React.FC;
export declare const Translate: React.FC;
export declare const InitImage: React.FC;
export declare const Error: React.FC;
//# sourceMappingURL=DropFile.stories.d.ts.map