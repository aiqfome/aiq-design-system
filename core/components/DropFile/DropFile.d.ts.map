{"version":3,"file":"DropFile.d.ts","sourceRoot":"","sources":["../../../src/components/DropFile/DropFile.tsx"],"names":[],"mappings":"AAAA,OAAO,KAA2B,MAAM,OAAO,CAAA;AAU/C,UAAU,aAAa;IACrB,OAAO,EAAE;QACP,IAAI,EAAE,MAAM,CAAA;QACZ,IAAI,EAAE,MAAM,CAAA;KACb,CAAA;IACD,SAAS,EAAE,MAAM,CAAA;IACjB,SAAS,EAAE,MAAM,CAAA;CAClB;AAED,MAAM,WAAW,KAAK;IACpB,WAAW,CAAC,EAAE,MAAM,CAAA;IACpB,QAAQ,CAAC,EAAE,CAAC,CAAC,EAAE,GAAG,KAAK,IAAI,CAAA;IAC3B,SAAS,CAAC,EAAE,MAAM,CAAA;IAClB,IAAI,CAAC,EAAE,MAAM,CAAA;IACb,SAAS,CAAC,EAAE,aAAa,CAAA;IAEzB,KAAK,CAAC,EAAE,MAAM,GAAG,MAAM,CAAA;IACvB,QAAQ,CAAC,EAAE,MAAM,GAAG,MAAM,CAAA;IAC1B,YAAY,CAAC,EAAE,MAAM,CAAA;IACrB,SAAS,CAAC,EAAE,OAAO,CAAA;CACpB;AAkDD,eAAO,MAAM,QAAQ,gFA+KpB,CAAA"}