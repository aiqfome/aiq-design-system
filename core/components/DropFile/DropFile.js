"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DropFile = void 0;
const react_1 = __importStar(require("react"));
const prop_types_1 = __importDefault(require("prop-types"));
const md_1 = require("react-icons/md");
const styled_components_1 = __importStar(require("styled-components"));
const Flex_1 = require("../Flex");
const Text_1 = require("../Text");
const Icon_1 = require("../Icon");
const translateDefault = {
    message: {
        row1: 'arraste e solte a imagem aqui',
        row2: 'ou clique para escolher'
    },
    errorSize: 'hey! a imagem é muito grande, escolha uma imagem menor',
    errorType: 'hey! apenas os tipos de png e jpeg são aceitos'
};
const Container = styled_components_1.default(Flex_1.Flex) `
  border: 1px solid
    ${({ error, theme }) => error ? theme.colors.red : theme.colors.mediumGrey};

  position: relative;
  &:hover {
    cursor: pointer;
  }

  ${({ isDragover }) => isDragover &&
    styled_components_1.css `
      background: #f5f5f5;
      border: 1px dashed #babcbe;
    `}
`;
const InputFile = styled_components_1.default.input `
  display: none;
`;
const Image = styled_components_1.default.img `
  width: calc(100% - 5%);
  height: 180px;

  border-radius: 6px;
  position: absolute;
`;
exports.DropFile = react_1.default.forwardRef((_a, ref) => {
    var { translate = translateDefault, dataMaxSize = 2048, initImage, name, errorMessage, errorForm, onChange } = _a, props = __rest(_a, ["translate", "dataMaxSize", "initImage", "name", "errorMessage", "errorForm", "onChange"]);
    const [isDragover, setIsDragover] = react_1.useState(false);
    const [image, setImage] = react_1.useState();
    const [typeError, setTypeError] = react_1.useState();
    const inputRef = react_1.useRef(document.createElement('input'));
    function handleDropFiles(e) {
        e.preventDefault();
        e.stopPropagation();
        setIsDragover(false);
        if (e.dataTransfer.files && e.dataTransfer.files[0]) {
            onChange && onChange(e.dataTransfer.files);
            inputRef.current.files = e.dataTransfer.files;
            setImageByFile(e.dataTransfer.files[0]);
        }
    }
    function handleClickBox() {
        inputRef.current.click();
    }
    function removeDragover(e) {
        e.preventDefault();
        e.stopPropagation();
        setIsDragover(false);
    }
    function addDragover(e) {
        e.preventDefault();
        e.stopPropagation();
        setIsDragover(true);
    }
    function handleChangeImage(e) {
        const input = e.target;
        onChange && onChange(input.files);
        if (input.files && input.files[0]) {
            setImageByFile(input.files[0]);
        }
    }
    function setImageByFile(file) {
        if (verifySize(file) && verifyType(file)) {
            const reader = new FileReader();
            reader.onload = e => {
                if (e.target) {
                    setImage(e.target.result);
                }
            };
            reader.readAsDataURL(file); // convert to base64 string
            return;
        }
        setImage(null);
    }
    function verifySize(file) {
        const kbytes = file.size / 1000;
        if (kbytes < dataMaxSize) {
            setTypeError(null);
            return true;
        }
        setTypeError('size');
        return false;
    }
    function verifyType(file) {
        if (file.type === 'image/jpeg' || file.type === 'image/png') {
            setTypeError(null);
            return true;
        }
        setTypeError('type');
        return false;
    }
    function getErrorMessage() {
        if (typeError === 'type') {
            return translate.errorType;
        }
        if (typeError === 'size') {
            return translate.errorSize;
        }
        if (errorForm) {
            return errorMessage;
        }
        return null;
    }
    const ImageFile = () => {
        if (initImage && !image && !isDragover) {
            return (react_1.default.createElement(Image, { width: '100%', height: '200px', src: initImage, alt: 'your image', "data-testid": 'dropfile-image' }));
        }
        if (image && !isDragover) {
            return (react_1.default.createElement(Image, { src: image, width: '100%', height: '200px', alt: 'your image', "data-testid": 'dropfile-image' }));
        }
        return react_1.default.createElement(Flex_1.Flex, null);
    };
    return (react_1.default.createElement(Container, Object.assign({ error: typeError || errorForm, isDragover: isDragover, onDragOver: addDragover, onDragEnter: addDragover, onDragLeave: removeDragover, onDragEnd: removeDragover, onDrop: handleDropFiles, onClick: handleClickBox, height: '200px', variant: 'centralized', borderRadius: 6, "data-testid": 'dropfile-container' }, props),
        react_1.default.createElement(InputFile, Object.assign({ name: name, onChange: handleChangeImage, ref: inputRef, type: 'file', "data-testid": 'dropfile-input' }, props)),
        react_1.default.createElement(Flex_1.Flex, { flexDirection: 'column', variant: 'centralized' },
            react_1.default.createElement(Icon_1.Icon, { color: 'grey', mb: '12px' },
                react_1.default.createElement(md_1.MdFileUpload, { size: 48 })),
            react_1.default.createElement(Text_1.Text, { color: 'grey' }, translate.message.row1),
            react_1.default.createElement(Text_1.Text, { color: 'grey' }, translate.message.row2),
            react_1.default.createElement(Text_1.Text, { mt: 4, color: 'red', fontWeight: 'medium', "data-testid": 'dropfile-error' }, getErrorMessage()),
            react_1.default.createElement(ImageFile, null))));
});
exports.DropFile.displayName = 'DropFile';
exports.DropFile.propTypes = {
    dataMaxSize: prop_types_1.default.number,
    translate: prop_types_1.default.any,
    onChange: prop_types_1.default.func,
    initImage: prop_types_1.default.string,
    errorMessage: prop_types_1.default.string,
    errorForm: prop_types_1.default.bool,
    name: prop_types_1.default.string
};
