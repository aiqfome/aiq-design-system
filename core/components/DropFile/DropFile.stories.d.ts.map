{"version":3,"file":"DropFile.stories.d.ts","sourceRoot":"","sources":["../../../src/components/DropFile/DropFile.stories.tsx"],"names":[],"mappings":"AAAA,OAAO,KAAuB,MAAM,OAAO,CAAA;;;;;AAM3C,wBAGC;AAED,eAAO,MAAM,KAAK,EAAE,KAAK,CAAC,EAMzB,CAAA;AAED,eAAO,MAAM,SAAS,EAAE,KAAK,CAAC,EAe7B,CAAA;AAED,eAAO,MAAM,SAAS,EAAE,KAAK,CAAC,EAkB7B,CAAA;AAED,eAAO,MAAM,KAAK,EAAE,KAAK,CAAC,EAiBzB,CAAA"}