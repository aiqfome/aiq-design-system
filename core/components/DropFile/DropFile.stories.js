"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Error = exports.InitImage = exports.Translate = exports.Basic = void 0;
const react_1 = __importDefault(require("react"));
const DropFile_1 = require("./DropFile");
const Flex_1 = require("../Flex");
exports.default = {
    component: DropFile_1.DropFile,
    title: 'DropFile'
};
exports.Basic = () => {
    return (react_1.default.createElement(Flex_1.Flex, { variant: 'fullCentralized' },
        react_1.default.createElement(DropFile_1.DropFile, { dataMaxSize: 2048, maxWidth: '680px', width: '100%' })));
};
exports.Translate = () => {
    const translate = {
        message: {
            row1: 'arraste e solte a imagem aqui',
            row2: 'ou clique para escolher'
        },
        errorSize: 'hey! a imagem é muito grande, escolha uma imagem menor',
        errorType: 'hey! apenas os tipos de png e jpeg são aceitos'
    };
    return (react_1.default.createElement(Flex_1.Flex, { variant: 'fullCentralized' },
        react_1.default.createElement(DropFile_1.DropFile, { translate: translate, maxWidth: '680px', width: '100%' })));
};
exports.InitImage = () => {
    function handleOnChangeFile(e) {
        console.log(e);
    }
    return (react_1.default.createElement(Flex_1.Flex, { variant: 'fullCentralized' },
        react_1.default.createElement(DropFile_1.DropFile, { dataMaxSize: 5000, initImage: 'https://cdn.startupi.com.br/wp-content/uploads/2019/12/Magalu-linx-marketplace.png', onChange: handleOnChangeFile, maxWidth: '680px', width: '100%' })));
};
exports.Error = () => {
    function handleOnChangeFile(e) {
        console.log(e);
    }
    return (react_1.default.createElement(Flex_1.Flex, { variant: 'fullCentralized' },
        react_1.default.createElement(DropFile_1.DropFile, { dataMaxSize: 5000, onChange: handleOnChangeFile, maxWidth: '680px', width: '100%', errorForm: true, errorMessage: 'Houston, we have a problem' })));
};
