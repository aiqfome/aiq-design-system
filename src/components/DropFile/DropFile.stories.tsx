import React, { ReactElement } from 'react'

import { DropFile } from './DropFile'

import { Flex } from '../Flex'

export default {
  component: DropFile,
  title: 'DropFile'
}

export const Basic: React.FC = (): ReactElement => {
  return (
    <Flex variant='fullCentralized'>
      <DropFile dataMaxSize={2048} maxWidth='680px' width='100%' />
    </Flex>
  )
}

export const Translate: React.FC = (): ReactElement => {
  const translate = {
    message: {
      row1: 'arraste e solte a imagem aqui',
      row2: 'ou clique para escolher'
    },
    errorSize: 'hey! a imagem é muito grande, escolha uma imagem menor',
    errorType: 'hey! apenas os tipos de png e jpeg são aceitos'
  }

  return (
    <Flex variant='fullCentralized'>
      <DropFile translate={translate} maxWidth='680px' width='100%' />
    </Flex>
  )
}

export const InitImage: React.FC = (): ReactElement => {
  function handleOnChangeFile(e) {
    console.log(e)
  }

  return (
    <Flex variant='fullCentralized'>
      <DropFile
        dataMaxSize={5000}
        initImage={
          'https://cdn.startupi.com.br/wp-content/uploads/2019/12/Magalu-linx-marketplace.png'
        }
        onChange={handleOnChangeFile}
        maxWidth='680px'
        width='100%'
      />
    </Flex>
  )
}

export const Error: React.FC = (): ReactElement => {
  function handleOnChangeFile(e) {
    console.log(e)
  }

  return (
    <Flex variant='fullCentralized'>
      <DropFile
        dataMaxSize={5000}
        onChange={handleOnChangeFile}
        maxWidth='680px'
        width='100%'
        errorForm={true}
        errorMessage='Houston, we have a problem'
      />
    </Flex>
  )
}
