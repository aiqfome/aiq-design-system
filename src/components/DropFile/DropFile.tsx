import React, { useState, useRef } from 'react'

import PropTypes from 'prop-types'

import { MdFileUpload } from 'react-icons/md'

import { Flex } from '../Flex'
import { Text } from '../Text'
import { Icon } from '../Icon'

import styled, { css, DefaultTheme } from 'styled-components'

interface TypeTranslate {
  message: {
    row1: string
    row2: string
  }
  errorSize: string
  errorType: string
}

export interface Props {
  dataMaxSize?: number
  onChange?: (e: any) => void
  value?: any
  initImage?: string
  translate?: TypeTranslate

  width?: number | string
  maxWidth?: number | string
}

const translateDefault = {
  message: {
    row1: 'arraste e solte a imagem aqui',
    row2: 'ou clique para escolher'
  },
  errorSize: 'hey! a imagem é muito grande, escolha uma imagem menor',
  errorType: 'hey! apenas os tipos de png e jpeg são aceitos'
}

interface ContainerProps extends DefaultTheme {
  error?: any
  isDragover?: boolean
}

const Container = styled(Flex)<ContainerProps>`
  border: 1px solid
    ${({ error, theme }) =>
      error ? theme.colors.red : theme.colors.mediumGrey};

  position: relative;
  &:hover {
    cursor: pointer;
  }

  ${({ isDragover }) =>
    isDragover &&
    css`
      background: #f5f5f5;
      border: 1px dashed #babcbe;
    `}
`

interface InputFileProps {
  ref: any
}

const InputFile = styled.input<InputFileProps>`
  display: none;
`

const Image = styled.img`
  width: calc(100% - 5%);
  height: 180px;

  border-radius: 6px;
  position: absolute;
`

export const DropFile: React.FC<Props> = ({
  translate = translateDefault,
  dataMaxSize = 2048,
  initImage,
  onChange,
  ...props
}) => {
  const [isDragover, setIsDragover] = useState(false)
  const [image, setImage] = useState<any>()
  const [typeError, setTypeError] = useState<null | string>()

  const inputRef = useRef(document.createElement('input'))

  function handleDropFiles(e) {
    e.preventDefault()
    e.stopPropagation()
    setIsDragover(false)

    if (e.dataTransfer.files && e.dataTransfer.files[0]) {
      inputRef.current.files = e.dataTransfer.files

      setImageByFile(e.dataTransfer.files[0])
    }
  }

  function handleClickBox() {
    inputRef.current.click()
  }

  function removeDragover(e) {
    e.preventDefault()
    e.stopPropagation()
    setIsDragover(false)
  }

  function addDragover(e) {
    e.preventDefault()
    e.stopPropagation()
    setIsDragover(true)
  }

  function handleChangeImage(e) {
    const input = e.target
    if (input.files && input.files[0]) {
      setImageByFile(input.files[0])
    }
  }

  function setImageByFile(file) {
    if (verifySize(file) && verifyType(file)) {
      onChange && onChange(file)
      const reader = new FileReader()

      reader.onload = e => {
        if (e.target) {
          setImage(e.target.result)
        }
      }

      reader.readAsDataURL(file) // convert to base64 string
      return
    }
    setImage(null)
  }

  function verifySize(file) {
    const kbytes = file.size / 1000
    if (kbytes < dataMaxSize) {
      setTypeError(null)
      return true
    }
    setTypeError('size')
    return false
  }

  function verifyType(file) {
    if (file.type === 'image/jpeg' || file.type === 'image/png') {
      setTypeError(null)
      return true
    }
    setTypeError('type')
    return false
  }

  function getErrorMessage() {
    if (typeError === 'type') {
      return translate.errorType
    }

    if (typeError === 'size') {
      return translate.errorSize
    }
    return null
  }

  const ImageFile: React.FC = () => {
    if (initImage && !image && !isDragover) {
      return (
        <Image height='200px' width='100%' src={initImage} alt='your image' />
      )
    }
    if (image && !isDragover) {
      return <Image height='200px' width='100%' src={image} alt='your image' />
    }
    return <Flex />
  }

  return (
    <Container
      error={typeError}
      isDragover={isDragover}
      onDragOver={addDragover}
      onDragEnter={addDragover}
      onDragLeave={removeDragover}
      onDragEnd={removeDragover}
      onDrop={handleDropFiles}
      onClick={handleClickBox}
      height='200px'
      variant='centralized'
      borderRadius={6}
      {...props}
    >
      <InputFile
        onChange={handleChangeImage}
        ref={inputRef}
        type='file'
        {...props}
      />
      <Flex flexDirection='column' variant='centralized'>
        <Icon color='grey' mb={'12px'}>
          <MdFileUpload size={48} />
        </Icon>
        <Text color='grey'>{translate.message.row1}</Text>
        <Text color='grey'>{translate.message.row2}</Text>
        <Text mt={4} color='red' fontWeight='medium'>
          {getErrorMessage()}
        </Text>
        <ImageFile />
      </Flex>
    </Container>
  )
}

DropFile.propTypes = {
  dataMaxSize: PropTypes.number,
  translate: PropTypes.any,
  onChange: PropTypes.func,
  initImage: PropTypes.string
}
