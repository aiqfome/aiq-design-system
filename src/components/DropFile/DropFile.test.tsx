import React from 'react'

import { DropFile } from './DropFile'

import { render } from '../utils/test/render'

describe('must match with the previous snapshot', () => {
  test('snapshot renders', () => {
    const component = render(<DropFile />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
