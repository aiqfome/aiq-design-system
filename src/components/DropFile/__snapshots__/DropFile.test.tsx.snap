// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`must match with the previous snapshot snapshot renders 1`] = `
<div
  className="sc-bdnylx kFqEoX sc-hKFyIo cxlIbw"
  height="200px"
  onClick={[Function]}
  onDragEnd={[Function]}
  onDragEnter={[Function]}
  onDragLeave={[Function]}
  onDragOver={[Function]}
  onDrop={[Function]}
>
  <input
    className="sc-eCApGN fA-Dehe"
    onChange={[Function]}
    type="file"
  />
  <div
    className="sc-bdnylx hgKHQl"
  >
    <div
      className="sc-bdnylx Bdjge sc-dlnjPT jJuwhT"
      color="grey"
    >
      <svg
        fill="currentColor"
        height={48}
        size={48}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width={48}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9 16h6v-6h4l-7-7-7 7h4zm-4 2h14v2H5z"
        />
      </svg>
    </div>
    <span
      className="sc-gtssRu gocbWW"
      color="grey"
    >
      arraste e solte a imagem aqui
    </span>
    <span
      className="sc-gtssRu gocbWW"
      color="grey"
    >
      ou clique para escolher
    </span>
    <span
      className="sc-gtssRu hUTPWP"
      color="red"
      fontWeight="medium"
    />
    <div
      className="sc-bdnylx ftODMf"
    />
  </div>
</div>
`;
